### PR TITLE
6470 hcm internal address config

### PR DESCRIPTION
**Pull request by ashishb-solo**
# Description

The new feature exposes `extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.InternalAddressConfig`

# Context

Users requested this feature in the linked issue

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/6470

### CODE REVIEW

File: projects/gloo/api/v1/options/hcm/hcm.proto [hunk 1]
```diff
@@ -218,4 +218,23 @@ message HttpConnectionManagerSettings {
...
+    // Manages Envoy's internal address configuration
+    message InternalAddressConfig {
+        // Whether unix socket addresses should be considered internal
+        bool unix_sockets = 1;
+
+        // List of CIDR ranges that are treated as internal
+        repeated CidrRange cidr_ranges = 2;
+    }
+
+    InternalAddressConfig internal_address_config = 38;
```
Comment (jackstine) [inline projects/gloo/api/v1/options/hcm/hcm.proto:239]: nit: add comments for `internal_address_config`

File: projects/gloo/pkg/plugins/hcm/plugin_test.go [hunk 2]
```diff
@@ -193,6 +206,33 @@ var _ = Describe("Plugin", func() {
...
+						PrefixLen:     &wrappers.UInt32Value{Value: 32},
+					},
+				},
+			},
+		}
+
+		settings.InternalAddressConfig.CidrRanges[0].AddressPrefix = "invalid_prefix"
+		cfg := &envoyhttp.HttpConnectionManager{}
+		err := processHcmNetworkFilter(cfg)
+		Expect(err).To(MatchError(ContainSubstring("invalid CIDR address")))
```
Comment (jackstine) [inline projects/gloo/pkg/plugins/hcm/plugin_test.go:235]: NICE!

**Review by jackstine — COMMENTED** [review]
LGTM! I'll keep it open for someone else to approve.

File: projects/gloo/api/v1/options/hcm/hcm.proto (hunk 1 above)
Comment (ashishb-solo) [reply projects/gloo/api/v1/options/hcm/hcm.proto:240]: nice catch! i addressed this in [744eb96](https://github.com/solo-io/gloo/pull/6817/commits/744eb96f81e1fa4fbc5ebb0f266698b3e686fb1d)

File: projects/sds/pkg/server/server.go [hunk 3]
```diff
@@ -122,7 +122,7 @@ func (s *Server) UpdateSDSConfig(ctx context.Context) error {
 	}
 	contextutils.LoggerFrom(ctx).Infof("Updating SDS config. sdsClient is %s. Snapshot version is %s", s.sdsClient, snapshotVersion)
 
-	secretSnapshot := cache.Snapshot{}
+	secretSnapshot := &cache.Snapshot{}
```
Comment (jenshu) [inline projects/sds/pkg/server/server.go:125]: why was this change needed?

File: projects/gloo/pkg/plugins/hcm/testing/expected_hcm_fields.json [hunk 4]
```diff
@@ -45,5 +45,7 @@
     "StripPortMode",
     "StreamErrorOnInvalidHttpMessage",
     "PathNormalizationOptions",
-    "StripTrailingHostDot"
-]
\ No newline at end of file
+    "StripTrailingHostDot",
+	"ProxyStatusConfig",
+	"TypedHeaderValidationConfig"
```
Comment (jenshu) [inline projects/gloo/pkg/plugins/hcm/testing/expected_hcm_fields.json:50]: do these fields come from the envoy dependency bump?
also nit: the indentation seems off

File: projects/sds/pkg/server/server.go (hunk 3 above)
Comment (ashishb-solo) [reply projects/sds/pkg/server/server.go:125]: See [here](https://github.com/solo-io/gloo/runs/7546296809?check_suite_focus=true) - the go-control-plane bump changed the interface for resource snapshot (see [here](https://github.com/envoyproxy/go-control-plane/commit/a51ba6fa54b2e107f5606a0e2799f8afed116c14) for the specific commit)

```
2022-07-27T18:18:47.9296287Z ##[error]error: failed to check packages: errors while loading package github.com/solo-io/gloo/projects/sds/pkg/server: [/home/runner/work/gloo/gloo/projects/sds/pkg/server/server.go:127:55: cannot use secretSnapshot (variable of type cache.Snapshot) as cache.ResourceSnapshot value in argument to s.snapshotCache.SetSnapshot: cache.Snapshot does not implement cache.ResourceSnapshot (method ConstructVersionMap has pointer receiver)]
```

File: projects/gloo/pkg/plugins/hcm/testing/expected_hcm_fields.json (hunk 4 above)
Comment (ashishb-solo) [reply projects/gloo/pkg/plugins/hcm/testing/expected_hcm_fields.json:50]: yes, they do. will fix the indentation

**Review by jenshu — APPROVED** [review]
LGTM!
